Sources/ProjectSpec/TargetSource.swift

```diff
@@ -100,12 +142,45 @@ extension TargetSource: JSONObjectConvertible {
         excludes = jsonDictionary.json(atKeyPath: "excludes") ?? []
         type = jsonDictionary.json(atKeyPath: "type")
         optional = jsonDictionary.json(atKeyPath: "optional") ?? false
-        if let string: String = jsonDictionary.json(atKeyPath: "buildPhase") {
-            if let buildPhase = BuildPhase(rawValue: string) {
-                self.buildPhase = buildPhase
-            } else {
-                throw SpecParsingError.unknownSourceBuildPhase(string)
+        
+        let copyFilesSettings: BuildPhase.CopyFilesSettings? = jsonDictionary.json(atKeyPath: "copyFiles")
+        let buildPhaseString: String? = jsonDictionary.json(atKeyPath: "buildPhase")
+        buildPhase = try BuildPhase(string: buildPhaseString, copyFilesSettings: copyFilesSettings)
+    }
+}
+
+extension TargetSource.BuildPhase {
+    
+    public init?(string: String?, copyFilesSettings: TargetSource.BuildPhase.CopyFilesSettings? = nil) throws {
+        let chosenString: String
+        if let string = string {
+            chosenString = string
+        } else if copyFilesSettings != nil {
+            chosenString = "copyFiles"
```

Comment:
The downside of this is that if they have a `TargetSource` with mixed files, there's no way to just specify the settings for the ones that will be in `copyFiles` by default, because if you do that everything will be moved to the `copyFiles` build phase.
I would be ok with keeping the `copyFiles` struct out of the enum like you had it before. The validation error could then be thrown in the generator when searching through the files and finding any copyFiles build phase `SourceFiles` (like modulemap) and it not having a `copyFiles` settings.

Are there other sorts of files you would want to add to `copyFiles`, or is it just `modulemap`?

Reply:
Header files in the case of static libraries.

Let me think about your downside comment. When I had it different before it would default the whole sources to the copy files build phase, and the way I see you recommending it is that _if_ it were determined to need to be copied it should use these settings. Right now I’m treating it exactly like all the other build phases and if specified they override any default logic. This phase just happens to require additional configuration options.